src/Sulu/Bundle/SecurityBundle/Entity/Role.php

```diff
@@ -139,10 +147,62 @@ public function removeGroup(Group $groups)
     /**
      * Get groups.
      *
-     * @return \Doctrine\Common\Collections\Collection
+     * @return Collection
      */
     public function getGroups()
     {
         return $this->groups;
     }
+
+    /**
+     * Add setting.
+     *
+     * @param RoleSetting $setting
+     *
+     * @return Role
+     */
+    public function addSetting(RoleSetting $setting)
+    {
+        $this->settings[] = $setting;
+
+        return $this;
+    }
+
+    /**
+     * Remove setting.
+     *
+     * @param RoleSetting $setting
+     */
+    public function removeSetting(RoleSetting $setting)
+    {
+        $this->settings->removeElement($setting);
+    }
+
+    /**
+     * Get settings.
+     *
+     * @return Collection
+     */
+    public function getSettings()
+    {
+        return $this->settings;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSetting($key)
+    {
+        $filteredSettings = $this->settings->filter(
+            function (RoleSetting $setting) use ($key) {
+                return $setting->getKey() === $key;
+            }
+        );
```

Comment:
fixed in my own branch. This could be done with "index-by" on the doctrine relation